File: 2847-find-maximum-number-of-string-pairs/find-maximum-number-of-string-pairs.py
class Solution:
    def maximumNumberOfStringPairs(self, words: List[str]) -> int:
        res = 0 
        for i in range(len(words)):
            for j in range(i+1,len(words)):
                k = words[j]
                if words[i]==k[::-1]:
                    res += 1
        return res
            

File: 2847-find-maximum-number-of-string-pairs/find-maximum-number-of-string-pairs.cpp
class Solution {
public:
    int maximumNumberOfStringPairs(vector<string>& words) {
        unordered_map<string ,int>mp ;
        int ans = 0  ;
        for(auto it: words){
            string rev = it ;
            reverse(rev.begin() , rev.end());
            if(mp[rev]>0)
            ans++ ;
            else
            mp[it]++;
        }
        return ans;
    }
};